### PR TITLE
Downgrade Mac version in CI/CD Pipeline

### PR DESCRIPTION
**Pull request by SmartManoj**
**What is the problem that this fixes or functionality that this introduces? Does it fix any open issues?**
Unit tests[[1]](https://github.com/OpenDevin/OpenDevin/actions/runs/9566040865/job/26370353142) [\[2\]](https://github.com/OpenDevin/OpenDevin/actions/runs/9569356473/job/26381733069?pr=2203) [3](https://github.com/OpenDevin/OpenDevin/actions/runs/9579624261/attempts/1?pr=2516) are also failing. https://github.com/abiosoft/colima/issues/1023#issuecomment-2120172847

### CODE REVIEW

**Review by neubig — DISMISSED** [review]
It looks like tests are still failing, and I also think we should try to make them work on the most recent version of Mac if possible

**Comment by SmartManoj** [comment]
Failed for 
```bash
chmod +x hello.sh
./hello.sh
```
Output:
```-bash: ./hello.sh: Permission denied
[Command -1 finished with exit code 126]
```

**Comment by tobitege** [comment]
There first permission error is already at the sandbox setup stage:
```log
14:52:44 - opendevin:WARNING: ssh_box.py:356 - Failed to chown workspace directory for opendevin in sandbox: b"chown: changing ownership of '/workspace': Permission denied\n". But this should be fine if the self.sandbox_workspace_dir='/workspace' is mounted by the app docker container.
```

**Comment by SmartManoj** [comment]
Is it constantly failing at the first attempt?

**Comment by tobitege** [comment]
> Is it constantly failing at the first attempt?

I wish I could answer that, I don't have a way of testing it for Darwin.

File: .github/workflows/run-integration-tests.yml [hunk 1]
```diff
@@ -0,0 +1,104 @@
+name: Run Integration Tests
```
Comment (li-boxuan) [inline .github/workflows/run-integration-tests.yml:1]: I assume this file is due to merge conflict resolution. Could you please remove this file? Then I am happy to approve this PR.

**Review by li-boxuan — APPROVED** [review]

**Pull request merged**